api/internal/loader/fileloader.go

```diff
@@ -169,7 +169,7 @@ func (fl *FileLoader) New(path string) (ifc.Loader, error) {
 	}
 	root, err := filesys.ConfirmDir(fl.fSys, fl.root.Join(path))
 	if err != nil {
-		return nil, errors.WrapPrefixf(err, ErrRtNotDir.Error()) //nolint:govet
```

Comment:
Could you update the go runtime version in this PR while keeping only the minimum necessary changes to pass existing tests and lint checks? If needed, we could extract the lint-related changes in this PR into a separate PR and merge that one first.

By keeping the go version bump PR simple, we'll make it easy to revert in case of any issues during upgrade.